### DISCUSSION TITLE
Calendar Localization French Challenge

### DISCUSSION THREAD

**Discussion by mouimet-infinisoft** [primefaces/community#468] [PrimeNG] [answered]
Hello everyone
We are trying to implement PrimeNG calendar component in French, unfortunately it's a failure. We are seeking for insight or anyone that could help as we spent many hours with no results. Here is a high level list of what's been done.

*Strategies Undertaken:*
1. **Reference to Documentation:**
   - Overview: Consultation of official documentation and recommended steps.
   - Outcome: Not working.

2. **Translation Service Implementation:**
   - Overview: Integration of a translation service to facilitate language translation within the Calendar component.
   - Outcome: Not working.

3. **Use of ngx Translation:**
   - Overview: Exploration of ngx translation as a potential solution.
   - Outcome: Not working.

4. **Asset Download Attempt:**
   - Overview: Direct download of assets from the PrimeNG website for translation.
   - Outcome: Not working.

5. **Label Override Efforts:**
   - Overview: Attempting to override Calendar component labels for a customized display language.
   - Outcome: Not working.
   - 
6. **Component Replacement:**
   - Overview: Removing the Calendar from PrimeNG and exploring the installation of a full calendar component as a replacement.
   - Outcome: Not working.

**Comment by LouLeGrain** [comment, marked as answer]
I managed to get it to work with ngx-translate v15, Angular 16.2.8, PrimeNG 16.0.2

![primeNg_calendar_fr](https://github.com/primefaces/community/assets/9808953/0a1b4d10-ea46-4a26-9ac0-4d57ec67252b)

There's plenty of documented ways to implement it

- [This medium article, wrote by a PrimeTek employee, it evens does it in french](https://yigitfindikli.medium.com/primeng-i18n-api-usage-with-ngx-translate-c5aa239937c8)
- [This primeng-lts documentation](https://www.primefaces.org/primeng-v14-lts/i18n)
- [This SO detailed response](https://stackoverflow.com/a/62874728/4855921)
- and so on
